packages/react-dom/src/events/ReactBrowserEventEmitter.js

```diff
@@ -84,22 +85,39 @@ import isEventSupported from './isEventSupported';
  *    React Core     .  General Purpose Event Plugin System
  */
 
-const alreadyListeningTo = {};
-let reactTopListenersCounter = 0;
+const elementListeningObjects: WeakMap<
+  Document | Element | Node,
+  ElementListeningObject,
+> = new WeakMap();
 
-/**
- * To ensure no conflicts with other potential React instances on the page
- */
-const topListenersIDKey = '_reactListenersID' + ('' + Math.random()).slice(2);
+// We store legacy events we listen where we don't use
+// passive/non-passive options on the event listener.
```

Comment:
It seems like those legacy events are also stored in the `ElementListeningObject`. Did I miss something or is this comment outdated?

Reply:
We store legacy and nonLegacy in the same object, which goes into the Map.

Reply:
Makes sense. I think I was confused by:

> We store [..] on the event listener.

Reply:
I've updated the comment to be more clear.